### PR TITLE
Add openzeppelin audit report to the repo

### PR DESCRIPTION
**Pull request by mmv08**
This PR:
- Implements https://github.com/safe-global/safe-modules/issues/233

### CODE REVIEW

**Review by remedcu — APPROVED** [review]

File: modules/4337/docs/v0.2.0/audit.md [hunk 1]
```diff
@@ -16,6 +18,22 @@ There is one acknowledged finding from the audit report:
...
+
+### Auditor
+
+OpenZeppelin (<https://openzeppelin.com/>).
+
+### Notes
+
+The final audit was performed on commit [3853f34f31837e0a0aee47a4452564278f8c62ba](https://github.com/safe-global/safe-modules/tree/3853f34f31837e0a0aee47a4452564278f8c62ba).
+
+No critical or high severity issues were found. For the detailed findings, please refer to the audit report.
```
Comment (nlordell) [inline modules/4337/docs/v0.2.0/audit.md:35]: We should mention that we did not deploy new contracts as a result of the audit review - as there were _no security-related smart contract code changes_.

File: modules/4337/docs/v0.2.0/audit.md [hunk 2]
```diff
@@ -16,6 +18,22 @@ There is one acknowledged finding from the audit report:
...
+
+### Auditor
+
+OpenZeppelin (<https://openzeppelin.com/>).
+
+### Notes
+
+The final audit was performed on commit [3853f34f31837e0a0aee47a4452564278f8c62ba](https://github.com/safe-global/safe-modules/tree/3853f34f31837e0a0aee47a4452564278f8c62ba).
+
+No security-related findings were reported, but we have implemented some of the suggested optimisations from the audit report into the codebase. These changes will be included in the next release of the 4337 module. For more information about the audit findings, please refer to the report.
```
Comment (nlordell) [inline modules/4337/docs/v0.2.0/audit.md:35]: I would reword this a bit - while there were security related findings (M-01, L-01 to 03), none of them led to actual Solidity outside of comments and pragmas.

Comment (mmv08) [reply modules/4337/docs/v0.2.0/audit.md:35]: Hmm, not sure I follow, for me security findings mean something that endager users' assets and require an immediate response. The issues you mentioned are about missing docstring, floating pragma and potential incompatibility. For me they're not really security related. Maybe we're looking for a better term.

Comment (nlordell) [reply modules/4337/docs/v0.2.0/audit.md:35]: > security findings mean something that endanger users' assets and require an immediate response

Personally, I tend to agree, but OZ audit qualified M-01 and L-0[1-3] as "security issues" in the report.

**Edit**: Nevermind, they are just "issues". I would still reword this to say that no issues were found that led to Solidity code changes.

**Review by nlordell — APPROVED** [review]
